### PR TITLE
Integrate pygbif and iucn api

### PR DESCRIPTION
**Pull request by dimasciput**
PR for issues : #93,  https://github.com/kartoza/healthyrivers/issues/94

### CODE REVIEW

File: django_project/fish/models/iucn_status.py [hunk 1]
```diff
@@ -4,29 +4,50 @@
...
         verbose_name_plural = 'IUCN Status'
         verbose_name = 'IUCN Status'
+
+
+@receiver(models.signals.pre_save, sender=IUCNStatus)
+def iucn_status_pre_save_handler(sender, instance, **kwargs):
+    if instance.category:
+        # if the category is Critically Endangered or Endangered or
+        # Vulnerable then the iucn status is sensitive
+        if instance.category == 'CR' or \
```
Comment (meomancer) [inline django_project/fish/models/iucn_status.py]: maybe we  make list for this on model class?
SENSITIVE_STATUS = ['CR','EN','VU']
or maybe giving sensitive status as third element on CATEGORY_CHOICES

File: django_project/base/utils/gbif.py [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+        ))
+        try:
+            response = species.name_usage(key=taxon.gbif_id)
+            if response:
+                taxon.common_name = response['canonicalName']
+                taxon.scientific_name = response['scientificName']
+                taxon.author = response['authorship']
+                taxon.save()
+                print('Taxon updated')
+        except HTTPError as e:
```
Comment (meomancer) [inline django_project/base/utils/gbif.py]: have you make sure there is no KeyError raised?

File: django_project/base/utils/gbif.py [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+    """
+    print('Find species : %s' % original_species_name)
+    list_of_species = []
+    try:
+        response = species.name_lookup(
+            q=original_species_name,
+            limit=3,
+            offset=2
+        )
+        results = response['results']
```
Comment (meomancer) [inline django_project/base/utils/gbif.py]: have you make sure there is no KeyError raised?

File: django_project/base/utils/gbif.py [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+    Update taxon for a fish collection.
+    :param fish_collection: Fish collection record model
+    """
+    results = find_species(fish_collection.original_species_name)
+
+    for result in results:
+        if 'nubKey' in result:
+            taxon, created = Taxon.objects.get_or_create(
+                    gbif_id=result['nubKey'])
+            taxon.common_name = result['canonicalName']
```
Comment (meomancer) [inline django_project/base/utils/gbif.py]: have you make sure there is no KeyError raised?

**Comment by dimasciput** [comment]
@meomancer thank you for the feedback, can you review it again?

**Comment by meomancer** [comment]
LGTM

**Pull request merged**